charts/tfy-agent/values.yaml

```diff
@@ -1,123 +1,296 @@
+  ## Annotations for tfyAgentProxy
+  ##
+  annotations: {}
 
-tfyAgentProxy:
-  # specifies if agent proxy should be enabled or not
-  enabled: true
-  imageTag: "00f2e71680781ab9e1e64270525b97d01e0ba236"
-  imageRepository: truefoundrycloud/tfy-agent-proxy
-  type: ClusterIP
+  ## Image of tfyAgentProxy
+  ##
+  image:
+    repository: truefoundrycloud/tfy-agent-proxy
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: "00f2e71680781ab9e1e64270525b97d01e0ba236"
 
-  # resource section for tfy agent proxy
+  ## Define resources requests and limits for single Pods.
+  ## ref: https://kubernetes.io/docs/user-guide/compute-resources/
+  ##
+  ## customized resource limits are very general in nature and must be kept the same
+  ## decreasing the resource might make the service UNAVAILABLE
   resources:
     limits:
-      cpu: '1'
+      cpu: 500m
+      memory: 512Mi
       ephemeral-storage: 500M
-      memory: 500M
     requests:
       cpu: 50m
+      memory: 128Mi
       ephemeral-storage: 200M
-      memory: 100M
-  # any additional env vars for tfy-agent-proxy
-  # can skip CLUSTER_TOKEN and CONTROL_PLANE_URL as they are read from .Value.env
-  env:
+  
+  ## SecurityContext holds pod-level security attributes and common container settings.
+  ## This defaults to non root user with uid 1000 and gid 2000. *v1.PodSecurityContext  false
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+  ##
+  securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+  ## If specified, the pod's tolerations.
+  ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+  ##
+  tolerations: []
+  # - key: "key"
+  #   operator: "Equal"
+  #   value: "value"
+  #   effect: "NoSchedule"
+    
+  ## Define which Nodes the Pods are scheduled on.
+  ## ref: https://kubernetes.io/docs/user-guide/node-selection/
+  ##
+  nodeSelector: {}
+
+  ## Assign custom affinity rules to the tfyAgent instance
+  ## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+  ##
+  affinity: {}
+  # nodeAffinity:
+  #   requiredDuringSchedulingIgnoredDuringExecution:
+  #     nodeSelectorTerms:
+  #     - matchExpressions:
+  #       - key: kubernetes.io/e2e-az-name
+  #         operator: In
+  #         values:
+  #         - e2e-az1
+  #         - e2e-az2
+
+
+ingress:
+  enabled: false
+  className: ""
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  hosts:
+    - host: chart-example.local
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+# Enabling virtual services for istio
+# disabled by default
+# required when tfy-agent is to be exposed outside
```

Comment:
We will not need to expose Agent outside. In the current architecture, we should ensure that the Agent is not contactable from the outside cluster apart form our control-plane, which does not require a VS.